lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -262,7 +292,7 @@ data ApiUtxoStatistics = ApiUtxoStatistics
 
 data WalletPostData = WalletPostData
     { addressPoolGap :: !(Maybe (ApiT AddressPoolGap))
-    , mnemonicSentence :: !(ApiMnemonicT '[15,18,21,24] "seed")
+    , mnemonicSentence :: !(ApiMnemonicT (AllowedMnemonics 'Shelley) "seed")
```

Comment:
The type is specific to Shelley indeed. We haven't specialized the name because pretty much **everything** that is Shelley-specific at the moment isn't specialized. That's an _unspoken convention_ of our codebase as far as I can tell... Types are Shelley-specific unless said differently explicitly. 

Renaming to 'ShelleyWalletPostData' makes sense, although it would induce a lot of other renaming elsewhere about many other types. 

Side-note: having 'Shelley' as part of the same 'WalletStyle' ADT feels kinda wrong (what would it mean to have a `ByronWalletPostData 'Shelley` ? I am still grooming this and I intended to open this PR as a draft, sorry.